src/amount_ext.rs

```diff
@@ -21,10 +25,53 @@ impl FromCliStr for Amount {
     }
 }
 
+fn parse_suffix(string: &str) -> Option<Result<Amount, anyhow::Error>> {
+    let final_value: f32;
```

Comment:
although we may be able to get away with floats here it's best to avoid any precision loss by just using the arithmetic in `Amount`.